packages/calendar/src/Calendar.ts

```diff
@@ -311,46 +383,193 @@ export class Calendar extends SpectrumElement {
-        this.selectedDate = calendarDate.toDate(this.timeZone);
+        this.currentDate = calendarDateEngaged;
+        this.selectedDate = calendarDateEngaged.toDate(this.timeZone);
 
         this.dispatchEvent(
             new CustomEvent('change', {
                 bubbles: true,
                 composed: true,
-                detail: this.selectedDate,
             })
         );
     }
 
+    private isClickInsideContentRadius(
+        event: MouseEvent,
+        element: HTMLElement
+    ): boolean {
+        const rect = element.getBoundingClientRect();
+        const radius = rect.width / 2;
+        const centerX = rect.left + radius;
+        const centerY = rect.top + radius;
+        const clickCenterDistance = Math.sqrt(
+            Math.pow(event.clientX - centerX, 2) +
+                Math.pow(event.clientY - centerY, 2)
+        );
+
+        return clickCenterDistance <= radius;
+    }
+
+    private handlePreviousMonth(): void {
+        const isSelectedInPreviousMonth =
+            this._selectedDate?.month === this.currentDate.month - 1;
+        const isTodayInPreviousMonth =
+            this.today.month === this.currentDate.month - 1;
+
+        if (isSelectedInPreviousMonth) this.currentDate = this._selectedDate!;
+        else if (isTodayInPreviousMonth) this.currentDate = this.today;
+        else
+            this.currentDate = startOfMonth(this.currentDate).subtract({
+                months: 1,
+            });
+
+        this.setWeeksInCurrentMonth();
+    }
+
+    private handleNextMonth(): void {
+        const isSelectedInNextMonth =
+            this._selectedDate?.month === this.currentDate.month + 1;
+        const isTodayInNextMonth =
+            this.today.month === this.currentDate.month + 1;
+
+        if (isSelectedInNextMonth) this.currentDate = this._selectedDate!;
+        else if (isTodayInNextMonth) this.currentDate = this.today;
+        else
+            this.currentDate = startOfMonth(this.currentDate).add({
+                months: 1,
+            });
+
+        this.setWeeksInCurrentMonth();
+    }
+
+    private handleKeydown(event: KeyboardEvent): void {
+        this.setDateFocusIntent();
+
+        const initialMonth = this.currentDate.month;
+
+        switch (event.code) {
+            case 'ArrowLeft': {
+                this.focusPreviousDay();
+                break;
+            }
+            case 'ArrowDown': {
+                this.focusNextWeek();
+                break;
+            }
+            case 'ArrowRight': {
+                this.focusNextDay();
+                break;
+            }
+            case 'ArrowUp': {
+                this.focusPreviousWeek();
+                break;
+            }
+            case 'Space':
+            case 'Enter': {
+                this.handleDaySelect(event);
+                break;
+            }
+        }
+
+        if (this.currentDate.month !== initialMonth)
+            this.setWeeksInCurrentMonth();
```

Comment:
This call, as well the ones from `handlePreviousMonth` and `handleNextMonth` will be removed from here and added in a special check inside `willUpdate`, I think this makes more sense and it's less error prone.